lib/dialects/oracle/connection-manager.js

```diff
@@ -0,0 +1,196 @@
+'use strict';
+
+var AbstractConnectionManager = require('../abstract/connection-manager')
+  , ConnectionManager
+  , Utils = require('../../utils')
+  , Promise = require('../../promise')
+  , sequelizeErrors = require('../../errors')
+  , dataTypes = require('../../data-types').oracle
+  , parserMap = {};
+  
+
+ConnectionManager = function(dialect, sequelize) {
+  AbstractConnectionManager.call(this, dialect, sequelize);
+
+  this.sequelize = sequelize;
+  this.sequelize.config.port = this.sequelize.config.port || 1521;
+  try {
+    if (sequelize.config.dialectModulePath) {
+      this.lib = require(sequelize.config.dialectModulePath);
+    } else {
+      this.lib = require('oracledb');
+      this.lib.maxRows = 500;
+      this.lib.Promise = require('bluebird');
+    }
+  } catch (err) {
+    if (err.code === 'MODULE_NOT_FOUND') {
+      throw new Error('Please install oracledb package manually');
+    }
+    throw err;
+  }
+
+  this.refreshTypeParser(dataTypes);
+};
+
+/**
+ * Method for checking the config object passed and generate the full database if not fully passed
+ */
+ConnectionManager.prototype.checkConfigObject = function(config) {
+    //A connectString should be defined
+    if(config.database.length == 0) {
+      var errorToThrow = "The database cannot be blank, you must specify the database name (which correspond to the service name"
+      errorToThrow +=" or specify the full connectString with the format : HOST:[PORT]/SERVICE_NAME";
+      errorToThrow += "\n from tnsnames.ora : (HOST = mymachine.example.com)(PORT = 1521)(SERVICE_NAME = orcl)" 
+      throw new Error(errorToThrow);
+    }
+
+      //The connectString has a special format, we check it
+      //ConnectString format is : host:[port]/service_name
+      if(config.database.indexOf('/') == - 1) {
```

Comment:
You should follow our linting rules, spaces before and after brackets in `if`

Also this is a URL parsing scheme, it should be placed here https://github.com/sequelize/sequelize/blob/54556cc4f80513b34fb5ab6a5fd0866387f6c706/lib/sequelize.js#L122